small-multiple-column-timeline/smallMultiColumnTimeChart.js

```diff
@@ -0,0 +1,180 @@
+export function draw() {
+    let yScale = d3.scaleLinear();
+    let xScale0 = d3.scaleTime();
+    let xScale1 = d3.scaleBand();
+    let seriesNames = [];
+    let yAxisAlign = 'right';
+    const includeAnnotations = d => (d.annotate !== '' && d.annotate !== undefined); // eslint-disable-line
+    let annotate = false; // eslint-disable-line
+    const colourScale = d3.scaleOrdinal()
+        // .range(gChartcolour.lineWeb)
+        .domain(seriesNames);
+
+    function chart(parent) {
+        parent.selectAll('rect')
+            .data(d => d.columnData)
+            .enter()
+            .append('rect')
+            .attr('class', 'columns')
+            .attr('id', d => `date: ${d.date} value: ${d.value}`)
+            .attr('x', d => xScale0(d.date))
+            .attr('transform', () => `translate(${-xScale1.bandwidth() / 2},0)`)
+            .attr('width', () => xScale1.bandwidth())
+            .attr('y', d => yScale(Math.max(0, d.value)))
+            .attr('height', d => Math.abs(yScale(d.value) - yScale(0)))
+            .attr('fill', () => colourScale());
+
+        // add titles for each chart
+        parent.append('text')
+            .attr('class', 'chart-label')
+            .attr('dy', -5)
+            .text(d => d.name.toUpperCase());
+    }
+
+    chart.yScale = (d) => {
+        if (!d) return yScale;
+        yScale = d;
+        return chart;
+    };
+    chart.yAxisAlign = (d) => {
+        if (!d) return yAxisAlign;
+        yAxisAlign = d;
+        return chart;
+    };
+
+    chart.highlightNames = (d) => {
+        highlightNames = d;
+        return chart;
+    };
+    chart.seriesNames = (d) => {
+        seriesNames = d;
+        return chart;
+    };
+    chart.xScale0 = (d) => {
+        if (!d) return xScale0;
+        xScale0 = d;
+        return chart;
+    };
+
+    chart.xScale1 = (d) => {
+        if (!d) return xScale1;
+        xScale1 = d;
+        return chart;
+    };
+    chart.plotDim = (d) => {
+        if (!d) return window.plotDim;
+        window.plotDim = d;
+        return chart;
+    };
+    chart.rem = (d) => {
+        if (!d) return rem;
+        rem = d;
+        return chart;
+    };
+    chart.annotate = (d) => {
+        annotate = d;
+        return chart;
+    };
+
+    chart.colourPalette = (d) => {
+        if (d === 'social' || d === 'video') {
+            colourScale.range(gChartcolour.lineSocial);
+        } else if (d === 'webS' || d === 'webM' || d === 'webMDefault' || d === 'webL') {
+            colourScale.range(gChartcolour.lineWeb);
+        } else if (d === 'print') {
+            colourScale.range(gChartcolour.linePrint);
+        }
+        return chart;
+    };
+
+    return chart;
+}
+
+export function drawHighlights() {
+    let yScale = d3.scaleLinear();
+    let xScale0 = d3.scaleTime();
+    let xScale1 = d3.scaleBand();
+    let invertScale = false; // eslint-disable-line
+
+    function highlights(parent) {
+        const highlights = parent.append('rect')
```

Comment:
`highlights` here shadows a variable in an upper scope and then is scoped only to the the closure it's used it in. In instances like this, it's probably better not to assign the result to a variable that never gets used.